packages/design-system/src/components/VerticalNav/VerticalNavItem.tsx

```diff
@@ -1,137 +1,195 @@
-    }
-  }
+export type VerticalNavItemComponent = React.ReactElement<any> | any | ((...args: any[]) => any);
 
+export interface VerticalNavItemProps {
   /**
-   * Called when VerticalNavItemLabel is clicked. Since the "label" could be
-   * a link, subnav toggle button, or plain text, we use this method to
-   * determine what action to take and which event to actually fire.
-   * @param {Object} SyntheticEvent
+   * @hide-prop This gets passed through from the parent VerticalNav to a nested VerticalNav
    */
-  handleLabelClick(evt) {
-    if (this.hasSubnav()) {
-      return this.handleToggleClick();
-    }
+  _selectedId?: string;
+  /**
+   * Aria label for the toggle button when the sub-navigation is collapsed
+   */
+  ariaCollapsedStateButtonLabel?: string;
+  /**
+   * Aria label for the toggle button when the sub-navigation is expanded
+   */
+  ariaExpandedStateButtonLabel?: string;
+  /**
+   * Additional classes to be added to the root element
+   */
+  className?: string;
+  /**
+   * When provided, this will render the passed in component. This is useful when
+   * integrating with React Router's `<Link>` or using your own custom component.
+   */
+  component?: VerticalNavItemComponent;
+  /**
+   * Whether or not the item's sub-navigation is in a collapsed state by default
+   */
+  defaultCollapsed?: boolean;
+  /**
+   * Called when the link is clicked, with the following arguments:
+   * [`SyntheticEvent`](https://facebook.github.io/react/docs/events.html),
+   * `id`, `url`.
+   * This takes precedence over the `VerticalNav` `onLinkClick` prop
+   */
+  onClick?: (...args: any[]) => any;
+  /**
+   * Called when this item's subnav is collapsed or expanded, with the
+   * following arguments: `id`, `collapsed`
+   */
+  onSubnavToggle?: (...args: any[]) => any;
+  /**
+   * Optional identifier. This can be handy if you're passing in an
+   * `onClick` handler. A unique ID will be generated if one isn't provided.
+   */
+  id?: string;
+  /**
+   * An array of nested `VerticalNavItem` data objects to be rendered in a
+   * sub-navigation list.
+   */
+  items?: any[];
+  /**
+   * Text to render for this nav item
+   */
+  label: React.ReactNode;
+  /**
+   * A URL to navigate to if this item is a link
+   */
+  url?: string;
+  /**
+   * If this item is currently selected
+   */
+  selected?: boolean;
+}
+
+export const VerticalNavItem = (props: VerticalNavItemProps): React.ReactElement => {
+  const id = props.id || uniqueId('VerticalNavItem_');
+  const subnavId = `${id}__subnav`;
 
-    return this.handleClick(evt);
-  }
+  const [collapsed, setCollapsed] = useState(props.defaultCollapsed);
+
+  // Refactor from `componentDidUpdate` to `isEffect` using this example:
+  // https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects
+  React.useEffect(() => {
+    if (props.onSubnavToggle) {
+      props.onSubnavToggle(props.id, collapsed);
+    }
+  }, [collapsed]);
```

Comment:
I don't actually know why this was in a `componentDidUpdate` in the first place and not just called in the `handleToggleClick` function. Do you know? That seems to be the only place we call `setCollapsed` anyway.

Reply:
Maybe at some point in its history, there were multiple things that could set the collapsed state, and it seemed cleaner to make a different function that called this whenever that state changed

Reply:
I tested it out in storybook, and I didn't notice a difference between just calling it at the end of `handleToggleClick` and doing it here.